ui/hooks/useDisplayName.ts

```diff
@@ -50,17 +55,21 @@ export function useDisplayNames(
 
     const name =
       nameEntry?.name ||
+      nftCollectionNameFromMetadata?.name ||
```

Comment:
The core reason for needing the NFT API and Reservoir for this is so we can only show default names for NFTs that are "trusted" meaning `isSpam` is `false`.

Reply:
Done https://github.com/MetaMask/metamask-extension/pull/25692/commits/6907abeee82d8814535fc3e1a630cef644cb8c05